xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -767,33 +770,42 @@ void CGUIDialogAddonInfo::BuildDependencyList()
 
     if (!addonInstalled)
     {
-      // after pushing the install button the dependency install dialog will
-      // pop up only if non-module dependencies are going to be installed or
-      // dependencies are unavailable. the latter is for informational purposes
-
-      if (showAllDependencies || !addonAvailable ||
-          addonAvailable->MainType() != ADDON_SCRIPT_MODULE)
+      // after pushing the install button the dependency install dialog
+      // will be opened only if...
+      // - dependencies are unavailable (for informational purposes) OR
+      // - the dependency is not a script/module                     OR
+      // - the script/module is not available at an official repo
+
+      if (!addonAvailable || addonAvailable->MainType() != ADDON_SCRIPT_MODULE ||
+          !CAddonRepos::IsFromOfficialRepo(addonAvailable, CheckAddonPath::NO))
       {
         m_showDepDialogOnInstall = true;
       }
     }
-
-    // AddonType ADDON_SCRIPT_MODULE needs to be filtered as these low-level add-ons
-    // should be hidden to the user in the dependency select dialog
-
-    if (showAllDependencies ||
-        (addonInstalled && addonInstalled->MainType() != ADDON_SCRIPT_MODULE) ||
-        (addonAvailable && addonAvailable->MainType() != ADDON_SCRIPT_MODULE) ||
-        (!addonAvailable && !addonInstalled))
+    else
     {
-      m_depsInstalledWithAvailable.emplace_back(dep, addonInstalled, addonAvailable);
+      // only display dialog if updates for already installed dependencies will install
+
```

Comment:
This feel off with the empty line between the comment and the line it refers to. Maybe an empty line before the comment and no empty line after?